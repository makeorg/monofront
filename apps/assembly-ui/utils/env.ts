const isClientSide = (): boolean => typeof window !== 'undefined';

// Define server env variables
const nodeEnv = (): string | undefined => process.env.NODE_ENV;
const isDev = (): boolean => nodeEnv() === 'development';
const contentUrlServerSide = (): string | undefined =>
  process.env.CONTENT_URL_SERVER_SIDE;
const assemblyUrlServerSide = (): string | undefined =>
  process.env.ASSEMBLY_URL_SERVER_SIDE;
const frontUrl = (): string | undefined => process.env.FRONT_URL;
const port = (): string | undefined => process.env.PORT;
const mixPanelToken = (): string | undefined => process.env.MIXPANEL_TOKEN;
const cookieFirstToken = (): string | undefined =>
  process.env.COOKIE_FIRST_TOKEN;
const fbPixelId = (): string | undefined => process.env.FB_PIXEL_ID;
const fbConversionToken = (): string | undefined =>
  process.env.FB_CONVERSION_TOKEN;
const apiUrlServerSide = (): string | undefined =>
  process.env.CORE_API_URL_SERVER_SIDE;
const apiUrlClientSide = (): string | undefined =>
  process.env.CORE_API_URL_CLIENT_SIDE;
const useLocalProxy = (): boolean => process.env.USE_LOCAL_PROXY === 'true';
const twPixelId = (): string => process.env.TW_PIXEL_ID || '';
const twAPIKey = (): string => process.env.TW_API_KEY || '';
const twAPISecret = (): string => process.env.TW_API_SECRET || '';
const twAccessToken = (): string => process.env.TW_ACCESS_TOKEN || '';
const twTokenSecret = (): string => process.env.TW_TOKEN_SECRET || '';
const toldToken = (): string => process.env.TOLD_TOKEN || '';

// Export in env object
export const env = {
  nodeEnv,
  isDev,
  contentUrlServerSide,
  assemblyUrlServerSide,
  frontUrl,
  port,
  isClientSide,
  mixPanelToken,
  cookieFirstToken,
  fbPixelId,
  fbConversionToken,
  apiUrlServerSide,
  apiUrlClientSide,
  useLocalProxy,
  twPixelId,
  twAPIKey,
  twAPISecret,
  twAccessToken,
  twTokenSecret,
  toldToken,
};
