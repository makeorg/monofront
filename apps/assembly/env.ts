const isClientSide = (): boolean => typeof window !== 'undefined';
const apiUrl = (): string | undefined => process.env.API_URL || window?.API_URL;
const port = (): string | undefined => process.env.PORT;

// Export in env object
export const env = {
  apiUrl,
  port,
  isClientSide,
};
