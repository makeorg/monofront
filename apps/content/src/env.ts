import 'dotenv/config';

// Define server env variables
const port = (): string | undefined => process.env.PORT;
const isProduction = (): boolean | undefined =>
  process.env.NODE_ENV === 'production';
const makeApiUrl = (): string | undefined => process.env.MAKE_API_URL;

// Export in env object
export const env = {
  port,
  isProduction,
  makeApiUrl,
};
