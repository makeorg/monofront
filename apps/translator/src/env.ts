import 'dotenv/config';

// Define server env variables
const port = (): string | undefined => process.env.PORT;
const deeplUrl = (): string | undefined => process.env.PROVIDER_DEEPL_URL;
const deeplKey = (): string | undefined => process.env.PROVIDER_DEEPL_AUTH_KEY;
const isProduction = (): boolean | undefined =>
  process.env.NODE_ENV === 'production';
const authEndpoint = (): string | undefined =>
  process.env.AUTHENTIFICATION_ENDPOINT;

// Export in env object
export const env = {
  port,
  deeplUrl,
  deeplKey,
  isProduction,
  authEndpoint,
};
