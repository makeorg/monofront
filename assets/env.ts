const onClientSide = typeof window !== 'undefined';

// Define client env variables
const nodeEnvWindow = () => onClientSide && window.NODE_ENV;
const isDevWindow = () => nodeEnvWindow === 'development';
const isTestWindow = () => nodeEnvWindow === 'ci';
const apiUrlWindow = () => onClientSide && window.API_URL;
const proxyTargetApiUrlWindow = () =>
  onClientSide && window.PROXY_TARGET_API_URL;
const frontUrlWindow = () => onClientSide && window.FRONT_URL;
const portWindow = () => onClientSide && window.PORT;

// Define server env variables
const nodeEnvProcess = () => process.env.NODE_ENV;
const isDevProcess = () => nodeEnvProcess === 'development';
const isTestProcess = () => nodeEnvProcess === 'ci';
const apiUrlProcess = () => process.env.API_URL;
const proxyTargetApiUrlProcess = () => process.env.PROXY_TARGET_API_URL;
const frontUrlProcess = () => process.env.FRONT_URL;
const portProcess = () => process.env.PORT;

// Export in env object
export const env = {
  nodeEnv: onClientSide ? nodeEnvWindow : nodeEnvProcess,
  isDev: onClientSide ? isDevWindow : isDevProcess,
  isTest: onClientSide ? isTestWindow : isTestProcess,
  apiUrl: onClientSide ? apiUrlWindow : apiUrlProcess,
  proxyTargetApiUrl: onClientSide
    ? proxyTargetApiUrlWindow
    : proxyTargetApiUrlProcess,
  frontUrl: onClientSide ? frontUrlWindow : frontUrlProcess,
  port: onClientSide ? portWindow : portProcess,
};
