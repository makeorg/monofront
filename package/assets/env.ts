const isClientSide = (): boolean => typeof window !== 'undefined';

// Define server env variables
const nodeEnv = (): string | undefined => process.env.NODE_ENV;
const isDev = (): boolean => nodeEnv() === 'development';
const isTest = (): boolean => nodeEnv() === 'test';
const apiUrlServerSide = (): string | undefined =>
  process.env.API_URL_SERVER_SIDE;
const apiUrlClientSide = (): string | undefined =>
  process.env.API_URL_CLIENT_SIDE;
const frontUrl = (): string | undefined => process.env.FRONT_URL;
const port = (): string | undefined => process.env.PORT;
const useLocalProxy = (): string | undefined => process.env.LOCAL_USE_PROXY;

// Facebook vars
const fbPixelId = (): string => process.env.FB_PIXEL_ID || '';
const fbConversionToken = (): string =>
  process.env.FB_CONVERSION_TOKEN || 'undefined';

// Hotjar vars
const hotjarToken = (): string => process.env.HOTJAR_TOKEN || '';
const contentApiUrlServerSide = (): string | undefined =>
  process.env.CONTENT_API_URL_SERVER_SIDE;

// Twitter vars
const twPixelId = (): string => process.env.TW_PIXEL_ID || '';
const twAPIKey = (): string => process.env.TW_API_KEY || '';
const twAPISecret = (): string => process.env.TW_API_SECRET || '';
const twAccessToken = (): string => process.env.TW_ACCESS_TOKEN || '';
const twTokenSecret = (): string => process.env.TW_TOKEN_SECRET || '';
const toldToken = (): string => process.env.TOLD_TOKEN || '';

// Export in env object
export const env = {
  nodeEnv,
  isDev,
  isTest,
  apiUrlServerSide,
  apiUrlClientSide,
  useLocalProxy,
  frontUrl,
  port,
  isClientSide,
  fbPixelId,
  fbConversionToken,
  hotjarToken,
  contentApiUrlServerSide,
  twPixelId,
  twAPIKey,
  twAPISecret,
  twAccessToken,
  twTokenSecret,
  toldToken,
};
