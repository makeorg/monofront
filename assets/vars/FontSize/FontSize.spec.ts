import React from 'react';
import renderer from 'react-test-renderer';
import { env } from 'Shared/env';
import { FontSize } from './index';

jest.mock('Shared/env');
env.frontUrl.mockReturnValue('http://test.make.org');

describe('Fonts', () => {
  it('must match the snapshot with default Props', () => {
    const component = renderer.create(<FontSize />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
