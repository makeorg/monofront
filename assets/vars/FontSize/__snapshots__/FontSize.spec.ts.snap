// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fonts must match the snapshot with default Props 1`] = `
Array [
  @media (min-width:360px) {

}

<h1>
    FontSize
  </h1>,
  .c0 {
  color: rgba(213,21,60,1);
  font-size: 12px;
}

.c0:hover,
.c0:focus {
  color: rgba(213,21,60,1);
}

.c1 {
  width: 14px;
  height: 14px;
  margin-left: 7px;
}

@media (min-width:360px) {
  .c0 {
    font-size: 16px;
  }
}

<table>
    <caption>
      Font size for the UI of
       
      <a
        className="c0"
        href="http://test.make.org"
      >
        http://test.make.org
      </a>
       
      from design tokens
       
      <a
        className="c0"
        href="https://www.npmjs.com/package/athena-design-tokens"
        rel="nooperner"
        target="_blank"
      >
        NPM module
        <svg
          aria-hidden={true}
          className="c1 red-link"
          height={12}
          viewBox="0 0 12 12"
          width={12}
        >
          <path
            className="tofill"
            d="M12.5.893v10.214c0 .77-.624 1.393-1.393 1.393H.893c-.77 0-1.393-.624-1.393-1.393V.893C-.5.123.124-.5.893-.5h10.214c.77 0 1.393.624 1.393 1.393zm-2.554.464H6.697c-.618 0-.93.75-.492 1.189l.928.928-5.674 5.674a.348.348 0 000 .492l.9.9a.347.347 0 00.493 0l5.674-5.673.928.928c.436.436 1.189.13 1.189-.492v-3.25a.697.697 0 00-.697-.696z"
            fillRule="evenodd"
          />
        </svg>
      </a>
       
    </caption>
    <thead>
      <tr>
        <th
          scope="col"
        >
          Size
        </th>
        <th
          scope="col"
        >
          Name
        </th>
        <th
          scope="col"
        >
          value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 56,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          X5L
        </td>
        <td>
          56
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 42,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          X4L
        </td>
        <td>
          42
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 36,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          X3L
        </td>
        <td>
          36
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 30,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          X2L
        </td>
        <td>
          30
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 26,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          XL
        </td>
        <td>
          26
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 22,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          L
        </td>
        <td>
          22
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 20,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          M
        </td>
        <td>
          20
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          S
        </td>
        <td>
          18
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          XS
        </td>
        <td>
          16
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <span
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            FontSize
          </span>
        </th>
        <td>
          X2S
        </td>
        <td>
          12
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
